src/System.Linq/tests/ShortCircuitingTests.cs

```diff
@@ -66,7 +69,10 @@ public void MinDoubleDoesntCheckAll()
             var tracker = new TrackingEnumerable(10);
             var source = tracker.Select(i => i == 5 ? double.NaN : (double)i);
             Assert.True(double.IsNaN(source.Min()));
-            Assert.Equal(5, tracker.Moves);
+
+            // .NET core shortcircuits as an optimization.
+            // See https://github.com/dotnet/corefx/pull/2350.
+            Assert.Equal(PlatformDetection.IsFullFramework ? 10 : 5, tracker.Moves);
```

Comment:
Yes, we have decided that it is ok to exit from "search" operators as soon as we have the result. We already had some inconsistencies where `First` would exit as soon as it found first thing, but `Single` would not exit as soon as it found a second. 
It is highly unlikely to have dependency on number of element/predicate accesses in such scenarios and did not seem worth preserving.

---
In reply to: [109671547](https://github.com/dotnet/corefx/pull/17877#discussion_r109671547) [](ancestors = 109671547)